spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/sql/TestCreateTable.java

```diff
@@ -341,4 +343,42 @@ public void testDowngradeTableToFormatV1ThroughTablePropertyFails() {
         "Cannot downgrade v2 table to v1",
         () -> sql("ALTER TABLE %s SET TBLPROPERTIES ('format-version'='1')", tableName));
   }
+
+  @Test
+  public void testCatalogSpecificWarehouseLocation() throws Exception {
+    File warehouseLocation = temp.newFolder(catalogName);
+    Assert.assertTrue(warehouseLocation.delete());
+    String location = "file:" + warehouseLocation.getAbsolutePath();
+    spark.conf().set("spark.sql.catalog." + catalogName + ".warehouse", location);
+    String testNameSpace =
+        (catalogName.equals("spark_catalog") ? "" : catalogName + ".") + "default1";
+    String testTableName = testNameSpace + ".table";
+    TableIdentifier newTableIdentifier = TableIdentifier.of("default1", "table");
+
+    sql("CREATE NAMESPACE IF NOT EXISTS %s", testNameSpace);
```

Comment:
Why do we need a custom namespace here?

Reply:
because `defaultWarehouseLocation(TableIdentifier tableIdentifier)` method will return exist namespace locationUri as warehouseLocation for HiveCatalog,  we can't change exist namespace location when set warehouseLocation, so i create a new namespace.

Reply:
? I don't follow. Does the current namespace have a location URI? I didn't think it had one. Does default just always have a uri?

Reply:
If the default always has a URI then by all means lets do this, also let's bump this into a "before all" method for the test suite, I don't think we need to call it for every test.